addons/website/static/src/components/dialog/dialog.xml

```diff
@@ -21,4 +21,69 @@
         </t>
     </Dialog>
 </t>
+
+<t t-name="website.AddPageTemplatePreview" owl="1">
+    <div t-ref="holder" class="gx-4 o_page_template" t-att-class="{ 'mb-4': props.firstRow }">
+        <div t-ref="preview" class="o_page_template_preview">
+            <iframe t-ref="iframe" src="about:blank"/>
+        </div>
+        <div class="o_button_area">
+            <button class="btn btn-primary" t-on-click="select">Use this template</button>
+        </div>
+    </div>
+</t>
+
+<t t-name="website.AddPageTemplatePreviews" owl="1">
+    <div class="container">
+        <div class="row gy-4">
+            <t t-foreach="[...this.columns.entries()]" t-as="column" t-key="column[0]">
+                <div class="col-lg-4 col-12">
+                    <t t-foreach="[...column[1].entries()]" t-as="template" t-key="template[0]">
+                        <AddPageTemplatePreview template="template[1]" firstRow="template_first"
+                            addPage="props.addPage" getCssLinkEls="props.getCssLinkEls" 
+                        />
+                    </t>
+                    <t t-if="column[1].length === 0">
+                        <div class="gx-4 o_page_template"/>
+                    </t>
+                </div>
+            </t>
+        </div>
+    </div>
+</t>
+
+<t t-name="website.AddPageTemplates" owl="1">
+    <div class="row py-4" t-att-class="{o_loading: state.loading}">
+        <Notebook orientation="'vertical'" pages="state.pages" onPageUpdate="(value) => this.onTemplatePageChanged(value)"/>
```

Comment:
All pages HTML is obtained in one single call. It is the creation of the `iframe` elements that trigger the subsequent requests for CSS, JS, fonts and images. 
I guess I can try to somehow "cache" the `iframe` elements so that when we navigate to a previously loaded page they are already loaded. (I.e. Detach them from the DOM before the page in unmounted so they don't get destroyed by Owl, and upon creating a new page, replace the empty iframes with the stored ones if they already exist.)

I do not think preloading all iframes would be a good idea:
- there is no limit to the number of templates that can be defined
- it would possibly slow down the loading of the ones that actually need to be displayed.

Also note that it loads way faster if we disable the mechanisms introduced to "wait for the page to be loaded":
```diff
@@ -104,7 +104,7 @@ export class AddPageTemplatePreview extends Component {
                 preloadLinkEl.setAttribute("rel", "preload");
                 preloadLinkEl.setAttribute("href", cssLinkEl.getAttribute("href"));
                 preloadLinkEl.setAttribute("as", "style");
-                iframeEl.contentDocument.head.appendChild(preloadLinkEl);
+//                iframeEl.contentDocument.head.appendChild(preloadLinkEl);
                 iframeEl.contentDocument.head.appendChild(cssLinkEl.cloneNode(true));
             }
             // Adjust styles.
@@ -164,10 +164,10 @@ export class AddPageTemplatePreview extends Component {
             // Make image loading eager.
             const lazyLoadedImgEls = wrapEl.querySelectorAll("img[loading=lazy]");
             for (const imgEl of lazyLoadedImgEls) {
-                imgEl.setAttribute("loading", "eager");
+//                imgEl.setAttribute("loading", "eager");
             }
             mainEl.appendChild(wrapEl);
-            await wUtils.onceAllImagesLoaded($(wrapEl));
+//            await wUtils.onceAllImagesLoaded($(wrapEl));
             // Restore image lazy loading.
             for (const imgEl of lazyLoadedImgEls) {
                 imgEl.setAttribute("loading", "lazy");
```
Sizes change a bit while loading, but you get a preview right away.